readme.md

```diff
@@ -0,0 +1,104 @@
+
+```java
+public final class MyPlugin implements Plugin<Project> {
+    public void apply(Project project) {
+        project.getPluginManager().apply(SuppressibleErrorPronePlugin.class);
+        
+        SuppressibleErrorProneExtension suppressibleErrorProneExtension =
+                project.getExtensions().getByType(SuppressibleErrorProneExtension.class);
+        
+        // Configure the base errorprone gradle plugin options in each sourceset
+        suppressibleErrorProneExtension.configureEachErrorProneOptions(errorProneOptions -> {
+            errorProneOptions.disable("CatchBlockLogException");
+            errorProneOptions.check("JavaxInjectOnAbstractMethod", CheckSeverity.WARNING);
+        });
+        
+        // You need to opt into which checks will apply suggested patches:
+        suppressibleErrorProneExtension.getPatchChecks().addAll(
+                "CollectionStreamForEach",
+                "ObjectsHashCodeUnnecessaryVarargs");
+        
+        // You can also conditionally opt into patch checks if a module is in the transitive
+        // runtimeClasspath of the source set:
+        suppressibleErrorProneExtension.getConditionalPathChecks().add(
+                new ConditionalPatchCheck(
+                        new IfModuleIsUsed("com.palantir.safe-logging", "preconditions"),
+                        "PreferSafeLoggingPreconditions",
+                        "PreferSafeLoggableExceptions")
+        );
+    }
+}
+```
+
+To actually suppress all the current failures, you need to run compilation twice:
+
+```
+./gradlew classes testClasses -PerrorProneSuppressStage1
+./gradlew classes testClasses -PerrorProneSuppressStage2
+```
+
+If rolling out automatically to lots of repos, we'd recommend running the fixes first before suppressing:
+
+```
+./gradlew classes testClasses -PerrorProneApply
+```
+
+You can also run fixes for individual checks:
+
+```
+./gradlew classes testClasses -PerrorProneApply=Check
+./gradlew classes testClasses -PerrorProneApply=Check,OtherCheck
+```
+
+Errorprone can be disabled by using the `-PerrorProneDisable` property.
+
+### Using directly from Gradle
+
+You can also use this directly in Gradle without making a plugin (not recommended if you are in a polyrepo environment):
+
+```gradle
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'com.palantir.suppressible-error-prone:gradle-suppressible-error-prone:<version>'
+    }
+}
+
+apply plugin: 'com.palantir.suppressible-error-prone'
+
+dependencies {
+    // You will actually need a source of errorprone checks, this include the default Google ones
+    errorprone 'com.google.errorprone:error_prone_core:<version>'
+}
+
+suppressibleErrorProne {
+    patchChecks.add('SomeCheck')
+}
+```
```

Comment:
all the info you put in this PRs description would be useful in this readme under a "How does this work?" section as well.

Reply:
Good point, I'll copy it over.